src/mscorlib/shared/Interop/Unix/System.Native/Interop.GetCwd.cs

```diff
@@ -25,29 +25,21 @@ internal static unsafe string GetCwd()
             }
 
             // If that was too small, try increasing large buffer sizes
-            // until we get one that works or until we hit MaxPath.
-            int maxPath = Interop.Sys.MaxPath;
-            if (StackLimit < maxPath)
+            int bufferSize = StackLimit;
+            do
             {
-                int bufferSize = StackLimit;
-                do
+                checked { bufferSize *= 2; }
+                var buf = new byte[bufferSize];
```

Comment:
You can change this to use `ArrayPool<byte>.Shared.Rent / Return` while you are on it.